### ISSUE TITLE
`jupyter lab build` always fails because of webpack 5.x and html-webpack-plugin 4.3.0

### ISSUE THREAD

**Issue by SchineCompton** [jupyterlab/jupyterlab#9649]
## Description

`jupyter lab build` always fails :

```
 The 'compilation' argument must be an instance of Compilation
```

I went into `staging` folder and find the `package.json` noticed the version specification

```
"devDependencies": {
...
        "html-webpack-plugin": "~4.3.0",
...
        "webpack": "^5.3.1",
...
    },
```

According to [issue](https://github.com/jantimon/html-webpack-plugin/issues/1451) from `html-webpack-plugin`, this problem can be fixed with `html-webpack-plugin@5.0.0-alpha.9` above.

I tried to manually change the version of `html-webpack-plugin` then it worked.

## Reproduce

1.  Install JupyterLab version: 3.0.5 on Anaconda distribution on (didn't test this on Linux)
2.  Run `jupyter lab build`

## Expected behavior

A successful build.

## Context

- Operating System and version: Windows 10
- JupyterLab version: 3.0.5

<details><summary>Troubleshoot Output</summary>
<pre>
$PATH:
	C:\Users\labuser\Anaconda3\envs\myenv
	C:\Users\labuser\Anaconda3\envs\myenv\Library\mingw-w64\bin
	C:\Users\labuser\Anaconda3\envs\myenv\Library\usr\bin
	C:\Users\labuser\Anaconda3\envs\myenv\Library\bin
	C:\Users\labuser\Anaconda3\envs\myenv\Scripts
	C:\Users\labuser\Anaconda3\envs\myenv\bin
	C:\Users\labuser\Anaconda3\condabin
	C:\Python36\Scripts
	C:\Python36
	C:\Windows\system32
	C:\Windows
	C:\Windows\System32\Wbem
	C:\Windows\System32\WindowsPowerShell\v1.0
	C:\Program Files\MATLAB\R2016a\runtime\win64
	C:\Program Files\MATLAB\R2016a\bin
	C:\Program Files (x86)\MiKTeX 2.9\miktex\bin
	C:\Program Files (x86)\IVI Foundation\VISA\WinNT\Bin
	C:\Program Files\IVI Foundation\VISA\Win64\Bin
	C:\Program Files (x86)\IVI Foundation\VISA\WinNT\Bin
	C:\Program Files\Microsoft SQL Server\110\Tools\Binn
	C:\Windows\SysWOW64\WindowsPowerShell\v1.0\Modules\TShell\TShell
	C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\bin
	C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC
	C:\Program Files (x86)\Windows Kits\10\Include\10.0.10240.0\ucrt
	C:\Program Files\Git\cmd
	C:\Windows\system32
	C:\Windows
	C:\Windows\System32\Wbem
	C:\Windows\System32\WindowsPowerShell\v1.0
	C:\Program Files\MATLAB\R2016a\runtime\win64
	C:\Program Files\MATLAB\R2016a\bin
	C:\Program Files (x86)\MiKTeX 2.9\miktex\bin
	C:\Program Files (x86)\IVI Foundation\VISA\WinNT\Bin
	C:\Program Files\IVI Foundation\VISA\Win64\Bin
	C:\Program Files (x86)\IVI Foundation\VISA\WinNT\Bin
	C:\Program Files\Microsoft SQL Server\110\Tools\Binn
	C:\Python27
	C:\Windows\SysWOW64\WindowsPowerShell\v1.0\Modules\TShell\TShell
	C:\Program Files (x86)\Microsoft Visual Studio 11.0\VC\bin
	C:\Program Files (x86)\Microsoft Visual Studio 11.0\VC
	C:\MinGW\bin
	C:\Qt\5.5\msvc2012\bin
	C:\Python27
	C:\Program Files (x86)\Labber\Script
	C:\Python27\PyQt-win-gpl-4.11.4
	C:\Python27\Lib\site-packages
	C:\Python27
	C:\Program Files (x86)\SumatraPDF
	C:\Users\labuser\nodejs

sys.path:
	C:\Users\labuser\Anaconda3\envs\myenv\Scripts\jupyter-troubleshoot.EXE
	C:\Code
	C:\Code\myenv_local\myenv\myenv-dev\control_software
	Z:\Code\myenv\myenv-user
	C:\oqc_innovate\python_libs\qiskit-terra-oqcmod
	C:\oqc_innovate\python_libs\qasm_interpreter_example
	C:\Program Files (x86)\Labber\Script
	c:\users\labuser\anaconda3\envs\myenv\python38.zip
	c:\users\labuser\anaconda3\envs\myenv\DLLs
	c:\users\labuser\anaconda3\envs\myenv\lib
	c:\users\labuser\anaconda3\envs\myenv
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages
	c:\users\labuser\control_software\myenv
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\oauthenticator-0.12.3-py3.8.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\tqdm-4.54.1-py3.8.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\pyqt5-5.15.2-py3.8-win-amd64.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\pyvisa_py-0.5.1-py3.8.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\scikit_optimize-0.8.1-py3.8.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\scikit_learn-0.24.0-py3.8-win-amd64.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\seaborn-0.11.1-py3.8.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\matplotlib-3.3.3-py3.8-win-amd64.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\pandas-1.2.0rc0-py3.8-win-amd64.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\scipy-1.6.0rc2-py3.8-win-amd64.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\numpy-1.20.0rc1-py3.8-win-amd64.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\pyqt5_sip-12.8.1-py3.8-win-amd64.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\pyaml-20.4.0-py3.8.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\joblib-1.0.0-py3.8.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\threadpoolctl-2.1.0-py3.8.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\pillow-8.0.1-py3.8-win-amd64.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\kiwisolver-1.3.1-py3.8-win-amd64.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\cycler-0.10.0-py3.8.egg
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\pyyaml-5.3.1-py3.8-win-amd64.egg
	c:\users\labuser\jupyterhub-winlocalprocessspawner
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\win32
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\win32\lib
	c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\Pythonwin

sys.executable:
	c:\users\labuser\anaconda3\envs\myenv\python.exe

sys.version:
	3.8.6 | packaged by conda-forge | (default, Dec 26 2020, 04:30:06) [MSC v.1916 64 bit (AMD64)]

platform.platform():
	Windows-10-10.0.10240-SP0

where jupyter:
	C:\Users\labuser\Anaconda3\envs\myenv\Scripts\jupyter.exe

	C:\Python36\Scripts\jupyter.exe

pip list:
	Package                           Version             Location

	--------------------------------- ------------------- --------------------------------------------------

	alembic                           1.4.3

	anyio                             2.0.2

	apptools                          5.1.0

	argon2-cffi                       20.1.0

	async-generator                   1.10

	attrs                             20.3.0

	Babel                             2.9.0

	backcall                          0.2.0

	bleach                            3.2.1

	certifi                           2020.12.5

	certipy                           0.1.3

	cffi                              1.14.4

	chardet                           4.0.0

	colorama                          0.4.4

	configobj                         5.0.6

	cryptography                      3.3.1

	cycler                            0.10.0

	decorator                         4.4.2

	defusedxml                        0.6.0

	entrypoints                       0.3

	envisage                          4.9.2

	gitdb                             4.0.5

	GitPython                         3.1.12

	idna                              2.10

	importlib-resources               5.1.0

	ipykernel                         5.4.2

	ipython                           7.19.0

	ipython-genutils                  0.2.0

	ipywidgets                        7.6.2

	jedi                              0.18.0

	Jinja2                            2.11.2

	joblib                            1.0.0

	json5                             0.9.5

	jsonschema                        3.2.0

	jupyter-client                    6.1.11

	jupyter-contrib-core              0.3.3

	jupyter-contrib-nbextensions      0.5.1

	jupyter-core                      4.7.0

	jupyter-highlight-selected-word   0.2.0

	jupyter-latex-envs                1.4.6

	jupyter-nbextensions-configurator 0.4.1

	jupyter-server                    1.2.1

	jupyter-telemetry                 0.1.0

	jupyterhub                        1.3.0

	jupyterhub-winlocalprocessspawner 1.0                 c:\users\labuser\jupyterhub-winlocalprocessspawner

	jupyterlab                        3.0.5

	jupyterlab-git                    0.23.3

	jupyterlab-pygments               0.1.2

	jupyterlab-server                 2.1.1

	jupyterlab-widgets                1.0.0

	kiwisolver                        1.3.1

	loguru                            0.5.3

	lxml                              4.6.2

	Mako                              1.1.3

	MarkupSafe                        1.1.1

	matplotlib                        3.3.3

	mayavi                            4.7.2

	mistune                           0.8.4

	mpld3                             0.5.2

	nbclassic                         0.2.6

	nbclient                          0.5.1

	nbconvert                         6.0.7

	nbdime                            2.1.0

	nbformat                          5.0.8

	nest-asyncio                      1.4.3

	notebook                          6.1.6

	numpy                             1.19.5

	oauthenticator                    0.12.3

	oauthlib                          3.1.0

	packaging                         20.8

	pandas                            1.2.0rc0

	pandocfilters                     1.4.3

	parso                             0.8.1

	pexpect                           4.8.0

	pickleshare                       0.7.5

	pillow                            8.0.1

	pip                               20.3.3

	prometheus-client                 0.9.0

	prompt-toolkit                    3.0.10

	psutil                            5.8.0

	ptyprocess                        0.7.0

	pyaml                             20.4.0

	pycparser                         2.20

	pyface                            7.2.0

	Pygments                          2.7.4

	pyOpenSSL                         20.0.1

	pyparsing                         2.4.7

	pypiwin32                         223

	PyQt5                             5.12.3

	PyQt5-sip                         4.19.18

	PyQtChart                         5.12

	PyQtWebEngine                     5.12.1

	pyrsistent                        0.17.3

	python-dateutil                   2.8.1

	python-editor                     1.0.4

	python-json-logger                2.0.1

	pytz                              2020.5

	PyVISA                            1.11.3

	pyvisa-py                         0.5.1

	pywin32                           300

	pywinpty                          0.5.7

	pyyaml                            5.3.1

	pyzmq                             20.0.0

	qutip                             4.5.2

	requests                          2.25.1

	ruamel.yaml                       0.16.12

	ruamel.yaml.clib                  0.2.2

	scikit-learn                      0.24.0

	scikit-optimize                   0.8.1

	scipy                             1.6.0

	seaborn                           0.11.1

	Send2Trash                        1.5.0

	setuptools                        51.3.3.post20210118

	six                               1.15.0

	smmap                             3.0.4

	sniffio                           1.2.0

	SQLAlchemy                        1.3.22

	terminado                         0.9.2

	testpath                          0.4.4

	threadpoolctl                     2.1.0

	tornado                           6.1

	tqdm                              4.54.1

	traitlets                         5.0.5

	traits                            6.1.1

	traitsui                          7.1.0

	typing-extensions                 3.7.4.3

	urllib3                           1.26.2

	vtk                               -PKG-VERSION

	wcwidth                           0.2.5

	webencodings                      0.5.1

	wheel                             0.36.2

	widgetsnbextension                3.5.1

	win32-setctime                    1.0.3

	wincertstore                      0.2

	zipp                              3.4.0

</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
[LabBuildApp] Building in c:\users\labuser\anaconda3\envs\myenv\share\jupyter\lab
[LabBuildApp] Node v15.3.0

[LabBuildApp] Yarn configuration loaded.
[LabBuildApp] Building jupyterlab assets (production, minimized)
[LabBuildApp] > node c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\jupyterlab\staging\yarn.js install --non-interactive
[LabBuildApp] yarn install v1.21.1
warning ..\..\..\..\..\..\..\package.json: No license field
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 0.84s.

[LabBuildApp] > node c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\jupyterlab\staging\yarn.js yarn-deduplicate -s fewer --fail
[LabBuildApp] yarn run v1.21.1
warning ..\..\..\..\..\..\..\package.json: No license field
$ c:\users\labuser\anaconda3\envs\myenv\share\jupyter\lab\staging\node_modules\.bin\yarn-deduplicate -s fewer --fail
Done in 0.88s.

[LabBuildApp] > node c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\jupyterlab\staging\yarn.js run build:prod:minimize
[LabBuildApp] yarn run v1.21.1
warning ..\..\..\..\..\..\..\package.json: No license field
$ webpack --config webpack.prod.minimize.config.js
[webpack-cli] [31mTypeError: The 'compilation' argument must be an instance of Compilation
    at Function.getCompilationHooks (c:\users\labuser\anaconda3\envs\myenv\share\jupyter\lab\staging\node_modules\webpack\lib\javascript\JavascriptModulesPlugin.js:119:10)
    at c:\users\labuser\anaconda3\envs\myenv\share\jupyter\lab\staging\node_modules\terser-webpack-plugin\dist\index.js:566:67
    at _next35 (eval at create (c:\Users\labuser\Anaconda3\envs\myenv\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:40:1)
    at _next13 (eval at create (c:\Users\labuser\Anaconda3\envs\myenv\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:97:1)
    at Hook.eval [as call] (eval at create (c:\Users\labuser\Anaconda3\envs\myenv\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:19:10), <anonymous>:127:1)
    at Hook.CALL_DELEGATE [as _call] (c:\Users\labuser\Anaconda3\envs\myenv\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:14:14)
    at Compiler.newCompilation (c:\Users\labuser\Anaconda3\envs\myenv\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:988:26)
    at c:\Users\labuser\Anaconda3\envs\myenv\share\jupyter\lab\staging\node_modules\webpack\lib\Compiler.js:1030:29
    at Hook.eval [as callAsync] (eval at create (c:\Users\labuser\Anaconda3\envs\myenv\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (c:\Users\labuser\Anaconda3\envs\myenv\share\jupyter\lab\staging\node_modules\tapable\lib\Hook.js:18:14)[39m
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[LabBuildApp] JupyterLab failed to build
[LabBuildApp] Traceback (most recent call last):

[LabBuildApp]   File "c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\jupyterlab\debuglog.py", line 47, in debug_logging
    yield

[LabBuildApp]   File "c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\jupyterlab\labapp.py", line 168, in start
    raise e

[LabBuildApp]   File "c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\jupyterlab\labapp.py", line 164, in start
    build(name=self.name, version=self.version,

[LabBuildApp]   File "c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\jupyterlab\commands.py", line 463, in build
    return handler.build(name=name, version=version, static_url=static_url,

[LabBuildApp]   File "c:\users\labuser\anaconda3\envs\myenv\lib\site-packages\jupyterlab\commands.py", line 675, in build
    raise RuntimeError(msg)

[LabBuildApp] RuntimeError: JupyterLab failed to build

[LabBuildApp] Exiting application: JupyterLab
</pre>
</details>

**Comment by jasongrout** [comment]
Thanks for tracking this down and testing a fix.

Glancing at that issue, it seems that it comes up when there are multiple versions of webpack in the tree. Perhaps that is why some can't reproduce it (presumably the multiple versions of webpack is a separate problem that may cause other problems).

I'm trying to see what the fix was in html-webpack-plugin, but I'm not seeing what commit it was.

Are you using the conda-forge package for JupyterLab?

This sounds like it is a duplicate of #9533.

**Comment by SachinCompton** [comment]
Ah, apologize for not noticing #9533. It looks like to be the same issue. When I change the `html-webpack-plugin` I also run `yarn install locally` in the `staging` folder, which could coincidentally hit the fix in that way.  

No I although it's an anaconda environment I just used pip to manage the package instead of conda.

**Comment by jasongrout** [comment]
Can you see why you have two versions of webpack in the default install? `yarn why webpack` in the staging folder will tell you who is requesting webpack and what versions.

**Comment by SachinCompton** [comment]
Thanks for pointing out!

This is what I have now

```
yarn why v1.21.1
warning ..\..\..\..\..\..\..\package.json: No license field
[1/4] Why do we have the module "webpack"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "webpack@5.14.0"
info Has been hoisted to "webpack"
info Reasons this module exists
   - Specified in "devDependencies"
   - Hoisted from "@jupyterlab#builder#webpack"
info Disk size without dependencies: "5.83MB"
info Disk size with unique dependencies: "9.63MB"
info Disk size with transitive dependencies: "22.7MB"
info Number of shared dependencies: 55
```

It seems now has only one webpack now. 

I have a customized extension which was build and installed before I upgrade to jupyterlab 3.x version. Then I found the problem in this issue and did `jupyter lab clean --all` to remove all the extensions. I thought that would give me a clean installation, but still have this problem. I suspect the other webpack version comes from the dependency of this extension. I now deleted all the version locks, rebuild this extension and install it on a new environment, it looked fine now.

Many thanks for all the help!

**Comment by jasongrout** [comment]
Diving a bit deeper, this commit in the html-webpack-plugin seems relevant: https://github.com/jantimon/html-webpack-plugin/commit/d722b9b35bec8b75ed71d934410607d71751f550

**Comment by jasongrout** [comment]
Upgrading html-webpack-plugin is now #9651

**Comment by kevinalexandr19** [comment]
Please I need help, I read through all the comments but I don't understand what exactly I must do to fix the compilation problem with webpack.

**Comment by kevinalexandr19** [comment]
@jasongrout please I need help, I dont understand the steps I must follow to update html-webpack-plugin and solve the build problem

**Comment by jasongrout** [comment]
The fix was released in jlab 3.0.6

**Comment by kevinalexandr19** [comment]
@jasongrout Hi, I just upgraded jupyterlab to 3.0.6 and installed node.js 15.7, then run `jupyter lab clean --all` followed by `jupyter lab build` but still got the same Compilation Error for webpack.  What is the sequence of steps I need to follow to solve the problem?

**Comment by jasongrout** [comment]
Can you follow the same sort of debugging exploration we did [above](https://github.com/jupyterlab/jupyterlab/issues/9649#issuecomment-764027858) to figure out why there are two copies of webpack in your build?

**Comment by kevinalexandr19** [comment]
> Can you follow the same sort of debugging exploration we did [above](https://github.com/jupyterlab/jupyterlab/issues/9649#issuecomment-764027858) to figure out why there are two copies of webpack in your build?

I don't have yarn installed, is that the problem maybe?

**Comment by jasongrout** [comment]
You can use `jlpm why` instead of `yarn why`

**Comment by jasongrout** [comment]
Or you can install `yarn` - in this case `yarn` is just parsing the `yarn.lock` file so we don't have to do it manually. It wouldn't be the cause of the problem.

**Comment by jasongrout** [comment]
You need to run it in the staging directory, in the jlab application directory (where the yarn.lock file is).

`jupyter lab path` gives the application directory. Go to the `staging` subdirectory of that directory, then run `jlpm why webpack`

**Comment by kevinalexandr19** [comment]
> You need to run it in the staging directory, in the jlab application directory (where the yarn.lock file is).
> 
> `jupyter lab path` gives the application directory. Go to the `staging` subdirectory of that directory, then run `jlpm why webpack`

This is the result now:
c:\Python\share\jupyter\lab\staging>jlpm why
yarn why v1.21.1
error Missing package name, folder or path to file to identify why a package has been installed
info Visit https://yarnpkg.com/en/docs/cli/why for documentation about this command.

This are the contents of staging:
![staging](https://user-images.githubusercontent.com/68877657/106198106-985f7400-6181-11eb-95be-34a40154f011.png)

**Comment by kevinalexandr19** [comment]
> You need to run it in the staging directory, in the jlab application directory (where the yarn.lock file is).
> 
> `jupyter lab path` gives the application directory. Go to the `staging` subdirectory of that directory, then run `jlpm why webpack`

Done, this is the result now:
c:\Python\share\jupyter\lab\staging>jlpm why
yarn why v1.21.1
error Missing package name, folder or path to file to identify why a package has been installed
info Visit https://yarnpkg.com/en/docs/cli/why for documentation about this command.

yarn why v1.21.1jupyter\lab\staging>jlpm why webpack
[1/4] Why do we have the module "webpack"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "webpack@5.18.0"
info Has been hoisted to "webpack"
info Reasons this module exists
   - Specified in "devDependencies"
   - Hoisted from "@jupyterlab#builder#webpack"
info Disk size without dependencies: "5.9MB"
info Disk size with unique dependencies: "9.7MB"
info Disk size with transitive dependencies: "22.77MB"
info Number of shared dependencies: 55
Done in 0.61s.

**Comment by kevinalexandr19** [comment]
> You need to run it in the staging directory, in the jlab application directory (where the yarn.lock file is).
> 
> `jupyter lab path` gives the application directory. Go to the `staging` subdirectory of that directory, then run `jlpm why webpack`

Done, this is the result now:
yarn why v1.21.1jupyter\lab\staging>jlpm why webpack
[1/4] Why do we have the module "webpack"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "webpack@5.18.0"
info Has been hoisted to "webpack"
info Reasons this module exists
   - Specified in "devDependencies"
   - Hoisted from "@jupyterlab#builder#webpack"
info Disk size without dependencies: "5.9MB"
info Disk size with unique dependencies: "9.7MB"
info Disk size with transitive dependencies: "22.77MB"
info Number of shared dependencies: 55
Done in 0.63s.

**Comment by kevinalexandr19** [comment]
> So this seems to indicate that only webpack 5 is installed?
> 
> It sounds like the problem may be different, or deeper. Can you open a new issue with the full error you are seeing? Please try to reproduce it in a clean environment so we can also try to reproduce it, and please fill out the issue template with the information it requests.

Ok, I'm on it.

**Comment by ilcarbo** [comment]
Having the same issue here, tried everything that was told in the thread.
my `yarn why webpack` shows
```
yarn why v1.21.1
[1/4] Why do we have the module "webpack"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "webpack@5.18.0"
info Has been hoisted to "webpack"
info Reasons this module exists
   - Specified in "devDependencies"
   - Hoisted from "@jupyterlab#builder#webpack"
info Disk size without dependencies: "5.9MB"
info Disk size with unique dependencies: "9.7MB"
info Disk size with transitive dependencies: "22.77MB"
info Number of shared dependencies: 55
```

already tried cleaning all, uninstalling and reinstalling and I can't seem to solve it. I'm going mad here, haha

**Comment by jasongrout** [comment]
See #9698 for some workarounds for the same error, but different cause. This other fix will be released in 3.0.7 soon.

**Comment by ilcarbo** [comment]
will take a look at them, thx.

**Comment by allenhutao** [comment]
> See [#9698 (comment)](https://github.com/jupyterlab/jupyterlab/issues/9698#issuecomment-771270053) for some workarounds for the same error, but different cause. This other fix will be released in 3.0.7 soon.

[LabBuildApp] Building in C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab
[LabBuildApp] Node v15.10.0

[LabBuildApp] Yarn configuration loaded.
[LabBuildApp] Building jupyterlab assets (production, minimized)
[LabBuildApp] > node C:\Users\lenovo\AppData\Roaming\Python\Python38\site-packages\jupyterlab\staging\yarn.js install --non-interactive
[LabBuildApp] yarn install v1.21.1
warning ..\..\..\..\..\..\..\package.json: No license field
[1/5] Validating package.json...
[2/5] Resolving packages...
success Already up-to-date.
Done in 1.91s.

[LabBuildApp] > node C:\Users\lenovo\AppData\Roaming\Python\Python38\site-packages\jupyterlab\staging\yarn.js yarn-deduplicate -s fewer --fail
[LabBuildApp] yarn run v1.21.1
warning ..\..\..\..\..\..\..\package.json: No license field
$ C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\.bin\yarn-deduplicate -s fewer --fail
Done in 1.55s.

[LabBuildApp] > node C:\Users\lenovo\AppData\Roaming\Python\Python38\site-packages\jupyterlab\staging\yarn.js run build:prod:minimize
[LabBuildApp] yarn run v1.21.1
warning ..\..\..\..\..\..\..\package.json: No license field
$ webpack --config webpack.prod.minimize.config.js
[webpack-cli] [31mModuleNotFoundError: Module not found: Error: Can't resolve '.\jquery.js' in 'C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\jquery\dist'
    at C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\webpack\lib\Compilation.js:1656:28
    at C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\webpack\lib\NormalModuleFactory.js:708:13
    at eval (eval at create (C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:10:1)
    at C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\webpack\lib\NormalModuleFactory.js:272:22
    at eval (eval at create (C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:9:1)
    at C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\webpack\lib\NormalModuleFactory.js:401:22
    at C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\webpack\lib\NormalModuleFactory.js:118:11
    at C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\webpack\lib\NormalModuleFactory.js:632:24
    at C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\webpack\lib\NormalModuleFactory.js:781:5
    at finishWithoutResolve (C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\enhanced-resolve\lib\Resolver.js:296:11)
resolve '.\jquery.js' in 'C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\jquery\dist'
  Parsed request is a module
  using description file: C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\jquery\package.json (relative path: ./dist)
    Field 'browser' doesn't contain a valid alias configuration
    resolve as module
      C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\jquery\dist\node_modules doesn't exist or is not a directory
      C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\jquery\node_modules doesn't exist or is not a directory
      C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\node_modules doesn't exist or is not a directory
      looking for modules in C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules
        single file module
          using description file: C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\package.json (relative path: ./node_modules/jquery.js)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\jquery.js doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\jquery.js.js doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\jquery.js.json doesn't exist
            .wasm
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\jquery.js.wasm doesn't exist
        C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\staging\node_modules\jquery.js doesn't exist
      C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\lab\node_modules doesn't exist or is not a directory
      C:\Users\lenovo\AppData\Roaming\Python\share\jupyter\node_modules doesn't exist or is not a directory
      C:\Users\lenovo\AppData\Roaming\Python\share\node_modules doesn't exist or is not a directory
      C:\Users\lenovo\AppData\Roaming\Python\node_modules doesn't exist or is not a directory
      C:\Users\lenovo\AppData\Roaming\node_modules doesn't exist or is not a directory
      C:\Users\lenovo\AppData\node_modules doesn't exist or is not a directory
      looking for modules in C:\Users\lenovo\node_modules
        single file module
          using description file: C:\Users\lenovo\package.json (relative path: ./node_modules/jquery.js)
            no extension
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\lenovo\node_modules\jquery.js doesn't exist
            .js
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\lenovo\node_modules\jquery.js.js doesn't exist
            .json
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\lenovo\node_modules\jquery.js.json doesn't exist
            .wasm
              Field 'browser' doesn't contain a valid alias configuration
              C:\Users\lenovo\node_modules\jquery.js.wasm doesn't exist
        C:\Users\lenovo\node_modules\jquery.js doesn't exist
      C:\Users\node_modules doesn't exist or is not a directory
      C:\node_modules doesn't exist or is not a directory[39m
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

[LabBuildApp] JupyterLab failed to build
[LabBuildApp] Traceback (most recent call last):

[LabBuildApp]   File "C:\Users\lenovo\AppData\Roaming\Python\Python38\site-packages\jupyterlab\debuglog.py", line 47, in debug_logging
    yield

[LabBuildApp]   File "C:\Users\lenovo\AppData\Roaming\Python\Python38\site-packages\jupyterlab\labapp.py", line 168, in start
    raise e

[LabBuildApp]   File "C:\Users\lenovo\AppData\Roaming\Python\Python38\site-packages\jupyterlab\labapp.py", line 164, in start
    build(name=self.name, version=self.version,

[LabBuildApp]   File "C:\Users\lenovo\AppData\Roaming\Python\Python38\site-packages\jupyterlab\commands.py", line 469, in build
    return handler.build(name=name, version=version, static_url=static_url,

[LabBuildApp]   File "C:\Users\lenovo\AppData\Roaming\Python\Python38\site-packages\jupyterlab\commands.py", line 678, in build
    raise RuntimeError(msg)

[LabBuildApp] RuntimeError: JupyterLab failed to build

[LabBuildApp] Exiting application: JupyterLab

**Comment by allenhutao** [comment]
> See [#9698 (comment)](https://github.com/jupyterlab/jupyterlab/issues/9698#issuecomment-771270053) for some workarounds for the same error, but different cause. This other fix will be released in 3.0.7 soon.

today i install some extension,then it failed 
i tried the methods aboved,but it still
need u help
